manifests/install.pp

```diff
@@ -18,29 +19,38 @@
   ipset::install::helper_script { ['ipset_sync', 'ipset_init']: }
 
   # autostart
-  if $::osfamily == 'RedHat' and $::operatingsystemmajrelease == '6' {
-    # upstart starter
-    file { '/etc/init/ipset.conf':
-      owner   => 'root',
-      group   => 'root',
-      mode    => '0644',
-      content => template("${module_name}/init.upstart.erb"),
-    }
-  } else {
-    if $::osfamily == 'RedHat' and $::operatingsystemmajrelease == '7' {
-      # systemd
-      file { '/usr/lib/systemd/system/ipset.service':
-        owner   => 'root',
-        group   => 'root',
-        mode    => '0644',
-        content => template("${module_name}/systemd.service.erb"),
+  case $::osfamily {
+    'RedHat', 'redhat': {
+      case $::operatingsystemrelease {
+        /^6.*/: { # upstart starter
+          file { '/etc/init/ipset.conf':
+            owner   => 'root',
+            group   => 'root',
+            mode    => '0644',
+            content => template("${module_name}/init.upstart.erb"),
+          }
+        }
+        /^7.*/: { # systemd
+          file { '/usr/lib/systemd/system/ipset.service':
+            owner   => 'root',
+            group   => 'root',
+            mode    => '0644',
+            content => template("${module_name}/systemd.service.erb"),
+          }
+          file { '/etc/systemd/system/basic.target.wants/ipset.service':
```

Comment:
I know this isn't your addition, but wouldn't it be better to just run `systemctl daemon-reload` as a refreshonly exec if the systemd file changes?  That way the backend will handle the linking incase someone decides to change the target at some point inside the template systemd service file... 
exec's name would be(to avoid conflicts) 'ipset_systemctl_daemonreload'  
I'd plug this stuff into a PR myself, but I just saw this module and was deciding whether to begin using it or not.